### PR TITLE
Adding test for snapshot of pvc with huge number of files

### PR DESCRIPTION
**Pull request by Avilir**
This PR adding a test which take snapshot for PVC with huge number of file on it  - 1M / 2M / 4M
an measure the time it tack.
the purpose of it, is to see if the number of file on the volume impact the time it take to create the snapshot.

Signed-off-by: Avi Liani <alayani@redhat.com>

### CODE REVIEW

**Review by jilju — COMMENTED** [review]
LGTM

File: tests/e2e/performance/test_pvc_snapshot_performance.py [hunk 1]
```diff
@@ -252,27 +301,174 @@ def test_pvc_snapshot_performance(self, teardown_factory, pvc_size):
...
+    )
+    def test_pvc_snapshot_performance_multiple_files(
+        self, ripsaw, file_size, files, threads, interface
+    ):
+        """
+        Run SmallFile Workload and the take snapshot.
+        test will run with 1M, 2M and 4M of file on the volume - total data set
+        is the same for all tests, ~30GiB, and then take snapshot and measure
+        the time it takes.
+        the test will run 3 time to check consistency.
```
Comment (vasukulkarni) [inline tests/e2e/performance/test_pvc_snapshot_performance.py:348]: docstring missing params explaination.

Comment (Avilir) [reply tests/e2e/performance/test_pvc_snapshot_performance.py:348]: fixed.

File: tests/e2e/performance/test_pvc_snapshot_performance.py [hunk 2]
```diff
@@ -252,27 +301,186 @@ def test_pvc_snapshot_performance(self, teardown_factory, pvc_size):
...
+                sleep=30,
+                timeout=600
+            )
+            for item in bench_pod.get()["items"][1]["spec"]["volumes"]:
+                if "persistentVolumeClaim" in item:
+                    pvc_name = item["persistentVolumeClaim"]["claimName"]
+                    break
+            log.info(f"Benchmark PVC name is : {pvc_name}")
+            # Creation of 4M files on CephFS can take a lot of time
+            timeout = 7200
```
Comment (keesturam) [inline tests/e2e/performance/test_pvc_snapshot_performance.py:445]: pls consider changing the timeout value based on the number of files

File: tests/e2e/performance/test_pvc_snapshot_performance.py [hunk 3]
```diff
@@ -82,55 +151,49 @@ def test_pvc_snapshot_performance(self, teardown_factory, pvc_size):
...
 
-        for test_num in range(tests_numbers):
+        for test_num in range(self.tests_numbers):
             test_results = {
-                'test_num': test_num + 1,
-                'dataset': (test_num + 1) * filesize * 1024,  # size in MiB
-                'create': {'time': None, 'speed': None},
-                'restore': {'time': None, 'speed': None},
+                "test_num": test_num + 1,
+                "dataset": (test_num + 1) * filesize * 1024,  # size in MiB
```
[the remark this answers is not in the document]
Comment (Avilir) [reply tests/e2e/performance/test_pvc_snapshot_performance.py:178]: the line `'test_num': test_num +1,` is needed in this dict for the final report.

File: tests/e2e/performance/test_pvc_snapshot_performance.py (hunk 2 above)
Comment (Avilir) [reply tests/e2e/performance/test_pvc_snapshot_performance.py:445]: There are more then one parameter (number of files) that influence the timeout : Number of files / File size / Block size / Volume type / Platform, and maybe more.
in most cases the test will not reach the timeout, and the purpose of the  workload is just to fill up the volume with files and the time it takes does not meter.

File: tests/e2e/performance/test_pvc_snapshot_performance.py [hunk 4]
```diff
@@ -55,12 +82,54 @@ def base_setup(
             status=constants.STATUS_RUNNING
         )
 
+    def create_snapshot(self, pvc_name, snap_name, interface):
```
[the remark this answers is not in the document]
Comment (Avilir) [reply tests/e2e/performance/test_pvc_snapshot_performance.py]: done.

**Comment by Avilir** [comment]
Closing and will submit new PR without conflicts

**Pull request closed without merging**